contracts/core/StakeManager.sol

```diff
@@ -43,17 +43,31 @@ abstract contract StakeManager is IStakeManager {
         depositTo(msg.sender);
     }
 
+
     /**
      * Increments an account's deposit.
      * @param account - The account to increment.
      * @param amount  - The amount to increment by.
      * @return the updated deposit of this account
      */
     function _incrementDeposit(address account, uint256 amount) internal returns (uint256) {
-        DepositInfo storage info = deposits[account];
-        uint256 newAmount = info.deposit + amount;
-        info.deposit = newAmount;
-        return newAmount;
+        unchecked {
+            DepositInfo storage info = deposits[account];
+            uint256 newAmount = info.deposit + amount;
+            info.deposit = newAmount;
+            return newAmount;
+        }
+    }
+
+    function _decrementDeposit(address account, uint256 amount) internal returns(bool) {
```

Comment:
Why returning bool when increment returns the new amount?

Reply:
@drortirosh I strongly suggest returning new value, reverting on underflow.

Reply:
reverting in this method requires StakeManager to know the FailedOp error, and also the opIndex - both are irrelevant for it.

Reply:
we return what is need.
too unfortunate that solidity can't return "err", like go. it would make "decrement" look better. returning amount doesn't help, since we only need to know if decrement caused an error (withdraw more than balance)
increment can't cause an error (as long as inputs are not out of bounds)

Reply:
At least rename to `_tryDecrementDeposit()`? The function doesn't always succeed